base/src/main/java/io/spine/code/js/FileName.java

```diff
@@ -120,6 +114,48 @@ public String pathToRoot() {
      * <p>Assumes that the path is used at the root directory with generated Protobuf files.
      */
     public String pathFromRoot() {
-        return CURRENT_DIR + value();
+        return ImportPath.currentDirectory() + value();
+    }
+
+    /**
+     * Tells whether the imported file was generated by Protobuf compiler.
+     */
+    public boolean isGeneratedProto() {
+        boolean result = value().endsWith(protoEnding());
+        return result;
+    }
+
+    /**
+     * Obtains the package name of the Protobuf file.
+     *
+     * @return the name of the package
+     * @throws IllegalStateException
+     *         if the file is not a file generated by Protobuf compiler
+     */
+    public PackageName protoPackage() {
```

Comment:
As long as I understand now, `FileName` is now about only the JS files generated from `.proto` sources. Let's say so in the docs. Also consider renaming `FileName`.